lib/fe/maybe.ex

```diff
@@ -139,4 +139,26 @@ defmodule FE.Maybe do
       end
     end)
   end
+
+  @doc """
+  Works like `fold/3`, except that the first element is converted to a `FE.Maybe`
+  value and passed as an initial second argument to the provided function.
+
+  ## Examples
+      iex> FE.Maybe.fold([1], fn _, _ -> FE.Maybe.nothing() end)
+      FE.Maybe.just(1)
+
+      iex> FE.Maybe.fold([1, 2, 3], &(FE.Maybe.just(&1 + &2)))
```

Comment:
I think you could make this the first example, because it shows the most simple scenario. Also, what do you think about adding an explanation to the currently first example? It took me a while to understand why it works that way :D

Reply:
In my mind, the first example show the most simple scenario, since a list of one is simpler than list of three :)
Or perhaps you mean, that's going to be most commonly used example?

I tried to make the description of the `Maybe.fold/2` more clear now, so that it's obvious why the first example works the way it works.
Does it help in any way?

Reply:
Yeah, I meant the most common scenario. But it's much better now with improved docs :)